test/test_mypy.py

```diff
@@ -0,0 +1,39 @@
+"""Tests using mypy."""
+
+import warnings
+import subprocess
+from os.path import join
+from pathlib import Path
+
+from qmflows.test_utils import PATH, Assertion_Warning
+from qmflows.backports import Literal
+
+_ROOT = Path(__file__).parts[:-2]
+PACKAGE = join(*_ROOT, 'src', 'qmflows')
+INI = str(PATH / 'mypy.ini')
+
+ACTION = frozenset({'raise', 'warn', 'ignore'})
+Action = Literal[tuple(ACTION)]  #: Type annotation for the 'action' keyword.
+
+
+def test_mypy(action: Action = 'warn') -> None:
```

Comment:
This is indeed a very fine addition to improve the code quality, I hope that we can soon enforce it!